code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -74,27 +75,28 @@
 	// WHY MUST WE DO THIS
 	// WHY
 	if(istype(Proj ,/obj/item/projectile/beam)||istype(Proj,/obj/item/projectile/bullet))
-		if(!istype(Proj ,/obj/item/projectile/beam/lastertag) && !istype(Proj ,/obj/item/projectile/beam/practice) && !Proj.nodamage)
+		if(!istype(Proj ,/obj/item/projectile/beam/lastertag) && !istype(Proj ,/obj/item/projectile/beam/practice) && !Proj.nodamage && !src.indestructible)
 			health -= Proj.damage
 	..()
-	if(health <= 0)
+	if(health <= 0 && !src.indestructible)
 		for(var/atom/movable/A as mob|obj in src)
 			A.loc = src.loc
 		del(src)
 	return
 
 /obj/item/weapon/storage/lockbox/ex_act(severity)
-	var/newsev = max(3,severity+1)
-	for(var/atom/movable/A as mob|obj in src)//pulls everything out of the locker and hits it with an explosion
-		A.loc = src.loc
-		A.ex_act(newsev)
-	newsev=4-severity
-	if(prob(newsev*25)+25) // 1=100, 2=75, 3=50
-		qdel(src)
+	if (!src.indestructible)
+		var/newsev = max(3,severity+1)
+		for(var/atom/movable/A as mob|obj in src)//pulls everything out of the locker and hits it with an explosion

```

Comment:
You don't need (as mob|obj)
That's a term only ever used for verbs and user inputs.